HarrisMoviesApplication/ViewController.swift

```diff
@@ -8,13 +8,42 @@
 import UIKit
 
 class ViewController: UIViewController {
+    
+    struct CatFact: Decodable {
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        // Do any additional setup after loading the view.
-        //creating
-    }
+        // MARK: - Properties
 
+        let text: String
+    }
+    
+    let factsArray:[CatFact] = []
 
+    @IBOutlet weak var catFactTableView: UITableView!
+    
+    override func viewDidLoad() {
+        let url = URL(string: "https://cat-fact.herokuapp.com/facts")!
+        
+            
+            let task = URLSession.shared.dataTask(with: url) { data, response, error in
+                if let data = data {
+                    if let facts = try? JSONDecoder().decode([CatFact].self, from: data) {
+                        print(facts)
+                    } else {
+                        print("Invalid Response")
+                    }
+                } else if let error = error {
+                    print("HTTP Request Failed \(error)")
+                }
+                
+                
+            }
+        
+        task.resume()
+        
+        catFactTableView.delegate = self
+        catFactTableView.dataSource = self
```

Comment:
Move this to the beginning of the function as this is a setup method we need done before anything else

Reply:
why would i do that? isn't the same?

Reply:
Isn't the same - We need these delegates to be set before we use the tableview later. The tableview doesn't work without the datasource

Reply:
oh it makes sense, ill check it thank you